plugin-server/src/worker/worker.ts

```diff
@@ -34,7 +34,21 @@ export async function createWorker(config: PluginsServerConfig, hub: Hub): Promi
 
             for (const signal of ['SIGINT', 'SIGTERM', 'SIGHUP']) {
                 if (updateJob) {
-                    process.on(signal, updateJob.cancel)
+                    process.on(signal, () => {
+                        try {
+                            updateJob.cancel()
+                        } catch (err) {
+                            if (
+                                err instanceof TypeError &&
+                                err.message.includes("Cannot read properties of undefined (reading 'length')")
+                            ) {
+                                // Ignore the annoying error node-schedule throws
```

Comment:
maybe a bit more context would be helpful - why does this error happen and why can we ignore it?

Reply:
The short answer is: I don't know but we already ignore it. :P (It only errors at process exit, FWIW.)

The long answer is I tried for a while to fix it before giving up, but I actually just had a thought that I'm going to try.

Reply:
Eh, I had no luck with my other idea.